src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -680,8 +695,154 @@ namespace OpenMS
       return EXECUTION_OK;
     }
 
-protected:
-    struct PeptideProteinMatchInformation
+     const String& getDecoy_string_() const;
+
+ protected:
+     using DecoyStringToAffixCount = std::map<std::string, std::pair<int, int>>;
+     using CaseInsensitiveToCaseSensitiveDecoy = std::map<std::string, std::string>;
+
+     template<typename T>
+     bool findDecoyString_(FASTAContainer<T>& proteins,
+                                std::string& decoy_string,
```

Comment:
I just realized that decoy_string and is_prefix are references to internal members. If you are not pulling out the function, you might as well just work directly on the members. Makes assignments in the code a bit clearer as well.